xmr-btc/src/bob.rs

```diff
@@ -1,28 +1,258 @@
 use crate::{
     alice,
     bitcoin::{
-        self, BroadcastSignedTransaction, BuildTxLockPsbt, SignTxLock, TxCancel,
-        WatchForRawTransaction,
+        self, poll_until_block_height_is_gte, BroadcastSignedTransaction, BuildTxLockPsbt,
+        SignTxLock, TxCancel, WatchForRawTransaction,
     },
     monero,
     monero::{CreateWalletForOutput, WatchForTransfer},
     serde::{bitcoin_amount, monero_private_key},
     transport::{ReceiveMessage, SendMessage},
 };
 use anyhow::{anyhow, Result};
+use async_trait::async_trait;
 use ecdsa_fun::{
     adaptor::{Adaptor, EncryptedSignature},
     nonce::Deterministic,
     Signature,
 };
+use futures::{
+    future::{select, Either},
+    FutureExt,
+};
+use genawaiter::sync::{Gen, GenBoxed};
 use rand::{CryptoRng, RngCore};
 use serde::{Deserialize, Serialize};
 use sha2::Sha256;
-use std::convert::{TryFrom, TryInto};
+use std::{
+    convert::{TryFrom, TryInto},
+    sync::Arc,
+    time::Duration,
+};
+use tokio::time::timeout;
+use tracing::error;
 
 pub mod message;
 pub use message::{Message, Message0, Message1, Message2, Message3};
 
+// TODO: Replace this with something configurable, such as an function argument.
+/// Time that Bob has to publish the Bitcoin lock transaction before both
+/// parties will abort, in seconds.
+pub const SECS_TO_ACT: u64 = 60;
```

Comment:
I agree with you. The value was chosen for the tests. Do you agree that it should be configurable? Also, this has to take into account the time it takes for Bob to achieve finality on the Bitcoin lock transaction, so it should depend on confirmations if we want to consider those at this stage.

Reply:
Making this configurable means that the application can control how much time it gives to Bob to lock the bitcoin. I moved the constant to the test module and adjusted it to be 5 seconds since it's just regtest in there.

We can choose the production value in the `swap` crate.

See: https://github.com/comit-network/xmr-btc-swap/pull/22/commits/328443aa5a441d5bd28c1f61437f28d9f63dca93

Reply:
Nice, thanks.